readthedocs/telemetry/collectors.py

```diff
@@ -0,0 +1,261 @@
+                    "channel": "conda-forge",
+                    "dist_name": "alabaster-0.7.12-py_0",
+                    "name": "alabaster",
+                    "platform": "noarch",
+                    "version": "0.7.12"
+                },
+                {
+                    "base_url": "https://conda.anaconda.org/conda-forge",
+                    "build_number": 0,
+                    "build_string": "pyh9f0ad1d_0",
+                    "channel": "conda-forge",
+                    "dist_name": "asn1crypto-1.4.0-pyh9f0ad1d_0",
+                    "name": "asn1crypto",
+                    "platform": "noarch",
+                    "version": "1.4.0"
+                }
+            ]
+        """
+        code, stdout, _ = self.run(
+            "conda", "list", "--json", "--name", self.version.slug
+        )
+        if code == 0 and stdout:
+            packages = self._safe_json_loads(stdout, [])
+            packages = [
+                {
+                    "name": package["name"],
+                    "channel": package["channel"],
+                    "version": package["version"],
+                }
+                for package in packages
+            ]
+            return packages
+        return []
+
+    def _get_pip_packages(self, include_all=False):
+        """
+        Get all the packages installed by the user using pip.
+
+        This includes top level and transitive dependencies.
+        The output of ``pip list`` is in the form of::
+
+            [
+                {
+                    "name": "requests-mock",
+                    "version": "1.8.0"
+                },
+                {
+                    "name": "requests-toolbelt",
+                    "version": "0.9.1"
+                },
+                {
+                    "name": "rstcheck",
+                    "version": "3.3.1"
+                },
+                {
+                    "name": "selectolax",
+                    "version": "0.2.10"
+                },
+                {
+                    "name": "slumber",
+                    "version": "0.7.1"
+                }
+            ]
+        """
+        cmd = ["python", "-m", "pip", "list", "--pre", "--local", "--format", "json"]
+        if not include_all:
+            cmd.append("--not-required")
+        code, stdout, _ = self.run(*cmd)
+        if code == 0 and stdout:
+            return self._safe_json_loads(stdout, [])
+        return []
+
+    def _get_operating_system(self):
+        """
+        Get the current operating system.
+
+        The output of ``lsb_release --description`` is in the form of::
+
+            Description:	Ubuntu 20.04.3 LTS
+        """
+        code, stdout, _ = self.run("lsb_release", "--description")
+        stdout = stdout.strip()
+        if code == 0 and stdout:
+            parts = stdout.split("\t")
+            if len(parts) == 2:
+                return parts[1]
+        return ""
```

Comment:
It would be good to use `None` values when the data is missing. This will make SQL queries easier, because we will able to check for `NULL` values instead.

Reply:
For strings not really strong opinion about using None or "", don't think this adds any complexity to the query